app/src/main/java/org/oppia/android/app/options/AppLanguageFragmentPresenter.kt

```diff
@@ -25,25 +31,58 @@ class AppLanguageFragmentPresenter @Inject constructor(private val fragment: Fra
       /* attachToRoot= */ false
     )
     this.prefSummaryValue = prefSummaryValue
-    languageSelectionAdapter = LanguageSelectionAdapter(prefKey) {
-      updateAppLanguage(it)
-    }
+    languageSelectionViewModel = getLanguageSelectionViewModel()
+    binding.viewModel = languageSelectionViewModel
+    languageSelectionViewModel.selectedLanguage.value = prefSummaryValue
     binding.languageRecyclerView.apply {
-      adapter = languageSelectionAdapter
+      adapter = createRecyclerViewAdapter()
     }
 
     binding.appLanguageToolbar?.setNavigationOnClickListener {
-      val message = languageSelectionAdapter.getSelectedLanguage()
+      val message = languageSelectionViewModel.selectedLanguage.value
       val intent = Intent()
       intent.putExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, message)
       (fragment.activity as AppLanguageActivity).setResult(REQUEST_CODE_APP_LANGUAGE, intent)
       (fragment.activity as AppLanguageActivity).finish()
     }
 
-    createAdapter()
     return binding.root
   }
 
+  fun getLanguageSelected(): String? {
+    return languageSelectionViewModel.selectedLanguage.value
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<LanguageItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
```

Comment:
Let's use a SingleTypeBuilder, instead, since we don't actually bind more than one type here. That should simplify this code a bit.

Ditto for audio language.

Reply:
Done